superchain/superchain_test.go

```diff
@@ -286,18 +287,20 @@ func TestContractBytecodes(t *testing.T) {
 // This is critical because the create2Deployer only activates on a block's timestamp.
 func TestCanyonTimestampOnBlockBoundary(t *testing.T) {
 	for superchainName, superchainConfig := range Superchains {
-		if superchainConfig.Config.CanyonTime == nil {
-			continue
+		for _, id := range superchainConfig.ChainIDs {
+			chainCfg := OPChains[id]
+			t.Run(fmt.Sprintf("%s_%s", superchainName, chainCfg.Name), testNetworkUpgradeTimestampOffset(chainCfg.Genesis.L2Time, 2, superchainConfig.Config.CanyonTime))
 		}
-		ct := *superchainConfig.Config.CanyonTime
+	}
+}
+
+// TestEcotoneTimestampOnBlockBoundary asserts that Ecotone will activate on a block's timestamp.
+// This is critical because the L2 upgrade transactions only activates on a block's timestamp.
+func TestEcotoneTimestampOnBlockBoundary(t *testing.T) {
```

Comment:
We could create a generic test function for this to avoid duplication in the future for standard chains, like
```go
func testStandardTimestampOnBlockBoundary(t *testing.T, desc string, ts func(Superchain) *uint64 ) {
	for superchainName, superchainConfig := range Superchains {
		for _, id := range superchainConfig.ChainIDs {
			chainCfg := OPChains[id]
			t.Run(fmt.Sprintf("%s/%s/%s", superchainName, chainCfg.Name, desc), testNetworkUpgradeTimestampOffset(chainCfg.Genesis.L2Time, 2, ts(superchainConfig))
		}
	}
}
```
and then call it for `Canyon` and `Ecotone`.